partner_ref_unique/models/res_partner.py

```diff
@@ -20,7 +19,6 @@ def _check_ref(self):
                 domain = [
                     ('id', '!=', partner.id),
```

Comment:
When you are creating a record, this condition shouldn't be added. I think this can be tested if `partner.id` is False.

Reply:
We have checked in an interactive session, and it's correct this way.